rotkehlchen/tasks/manager.py

```diff
@@ -547,13 +557,20 @@ def _schedule(self) -> None:
         if not_proceed:
             return  # too busy
 
-        callables = random.sample(
-            population=self.potential_tasks,
-            k=min(self.max_tasks_num - current_greenlets, len(self.potential_tasks)),
-        )
+        random.shuffle(self.potential_tasks)
+        max_tasks = min(self.max_tasks_num - current_greenlets, len(self.potential_tasks))
 
-        for callable_fn in callables:
-            callable_fn()
+        spawned_new = 0
+        for callable_fn in self.potential_tasks:
```

Comment:
Make it obvious these functions schedule. They are not the final task itself

```suggestion
        for scheduling_fn in self.potential_tasks:
```